pic32/cores/pic32/WInterrupts.c

```diff
@@ -263,12 +265,22 @@ void __attribute__((nomips16,at_vector(_EXTERNAL_0_VECTOR),interrupt(IPL4SRS)))
 void __attribute__((interrupt(),nomips16)) ExtInt0Handler(void)
 #endif
 {
-
+	VEC_ENT_CODE(EXT_INT0,INT0IF,PIN_INT0);
```

Comment:
Basically, clearing IFS early is an optimization. It allows the code to not have to check anything for edge cases. Furthermore it emulates exactly how it would look in hardware that has actual level IRQs. Yes, you want to exit the ISR so that other ISR on the same priority/subpriority get a chance to execute. By doing it this way you also avoid race conditions.